### PR TITLE
Use block style for multiline YAML values.

### PR DESCRIPTION
**Pull request by josepht**
Make the parts.yaml, 'search', and 'define' output nicer.

* Use block style for multiline strings
* Support OrderedDict for all yaml usage.
* Only show the first line of the description in 'search' output.

LP:#1614651

### CODE REVIEW

**Comment by sergiusens** [comment]
looks good. Mind opening a bug to move this logic and the OrderedDict one to the base of it all (`snapcraft.__init__.py) so mostly everything gets to use it?

**Comment by sergiusens** [comment]
Hmm, maybe instead of a bug, make it in this PR, there are two problems now

1. `snapcraft search` is not cutting off on new lines (it should and this PR exposes that problem).
2. `snapcraft define` shows the *ugly* output this PR is supposed to fix

Here is the run through (after generating the parts and copying it into `.local/share/snapcraft/parts.yaml`)

```
sergiusens@snappy:~/snapcraft$ snapcraft search desktop
PART NAME          DESCRIPTION
desktop-qt5        Helpers for qt5 minimal launchers.
It brings the necessary code and ex...
desktop-gtk3       Helpers for gtk3 minimal launchers.
It brings the necessary code and e...
desktop-gtk2       Helpers for gtk2 minimal launchers.
It brings the necessary code and e...
desktop-glib-only  Helpers for glib minimal launchers.
It brings the necessary code and e...
desktop-qt4        Helpers for qt4 minimal launchers.
It brings the necessary code and ex...
sergiusens@snappy:~/snapcraft$ snapcraft define desktop-qt5
Maintainer: 'Snapcraft community <snapcraft@lists.snapcraft.io>'
Description: 'Helpers for qt5 minimal launchers.\nIt brings the necessary code and exports for binding and using those\ndesktop technologies in a relocatable fashion, enabling binding with\nglobal desktop theme, icon theme, image caching, fonts, mimetype handlers\napplication global menu and gsettings integration.\nIt also brings basics ubuntu dependency packages.\n\nUsage :\n  1. add "after: [desktop-qt5]" to your launcher:\n  2. prepend your command with "desktop-launch", like:\n     commands: "desktop-launch foo" if foo is in /home/sergiusens/node/bin:/home/sergiusens/gems/bin:/home/sergiusens/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin:/home/sergiusens/source/launchpad.net/ubuntu-archive-tools. You can as well\n     specify: "desktop-launch /foo".\n  3. add needed plugs to your application:\n     - for graphical application:\n       plugs: [x11 (or unity7 for appmenu integration)]. Think about adding\n       opengl if you need hw acceleration.\n     - if your application needs access to sound:\n       plugs: [pulseaudio]\n     - accessing to user\'s home directory:\n       plugs: [home]\n     - read/write to gsettings:\n       plugs: [gsettings, home]\n       (note that the home plug is needed to read new value)\n'

desktop-qt5:
  build-packages:
  - qtbase5-dev
  - dpkg-dev
  make-parameters:
  - FLAVOR=qt5
  plugin: make
  source: https://github.com/ubuntu/snapcraft-desktop-helpers.git
  source-subdir: qt
  stage-packages:
  - libxkbcommon0
  - ttf-ubuntu-font-family
  - dmz-cursor-theme
  - light-themes
  - shared-mime-info
  - libqt5gui5
  - libgdk-pixbuf2.0-0
  - libqt5svg5
  - appmenu-qt5
```

File: snapcraft/tests/test_commands_define.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def test_defining_a_part_that_exists(self, mock_stdout):
         main.main(['define', 'curl'])
 
         expected_output = """Maintainer: 'none'
-Description: 'test entry for curl'
+Description: test entry for curl
```
Comment (sergiusens) [inline snapcraft/tests/test_commands_define.py]: can you add a multiline description test as well?

Comment (josepht) [inline snapcraft/tests/test_commands_define.py]: I added on in the latest commit.  Thanks for the suggestion.

Comment (sergiusens) [inline snapcraft/tests/test_commands_define.py]: oh, but for search as that is the one that had the problem ;-)
This is still good!

**Pull request merged**